image-accessible-name: improve examples (#1300)
act-rules/act-rules.github.io: _rules/image-accessible-name-23a2a8.md

```html
<div
	role="img"
	aria-label="W3C logo"
	style="width:72px; height:48px; background-image: url(/test-assets/shared/w3c-logo.png)"
></div>
```

#### Passed Example 3

This element with a [semantic role][] of `img` has an [accessible name][] because of an `aria-labelledby` attribute and an element with matching `id`.

```html
<div style="display: none" id="img-label">W3C logo</div>
<div
	role="img"
	aria-labelledby="img-label"
	style="width:72px; height:48px; background-image: url(/test-assets/shared/w3c-logo.png)"
></div>
```

#### Passed Example 4

This `img` element has an [accessible name][] because of a `title` attribute.

**Note**: There are assistive technologies that do not support using the `title` attribute for an [accessible name][], or in which this feature can be disabled.

```html
<img title="W3C logo" src="/test-assets/shared/w3c-logo.png" />
```

#### Passed Example 5

This `img` element has an [implicit role][] of `presentation` because of the empty `alt` attribute.

```html
<img alt="" src="/test-assets/shared/background.png" />
```

#### Passed Example 6

> The `img` element has an [explicit role][] of `presentation` because of the value of the `role` attribute.

Reviewer: ```suggestion
This `img` element has an [explicit role][] of `presentation` because of the value of the `role` attribute.
```

```html
<img role="presentation" style="width:72px; height:48px; background-image: url(/test-assets/shared/background.png)" />
```

#### Passed Example 7

> The `img` element has an [explicit role][] of `none` because of the value of the `role` attribute.

Reviewer: ```suggestion
This `img` element has an [explicit role][] of `none` because of the value of the `role` attribute.
```

```html
<img role="none" src="/test-assets/shared/background.png" />
```

#### Passed Example 8

This `img` element inside a `div` positioned off screen has an [implicit role][] of `img` because of the empty `alt` attribute.

```html
<div style="margin-left:-9999px;">
	<img alt="" src="/test-assets/shared/background.png" />
</div>
```

### Failed

#### Failed Example 1

This `img` element has an empty [accessible name][] and an [implicit role][] of `img` because it is missing an empty `alt` attribute.

```html
<img src="/test-assets/shared/w3c-logo.png" />
```

#### Failed Example 2

This element with role of `img` has an empty [accessible name][].

```html
<div role="img" style="width:72px; height:48px; background-image: url(/test-assets/shared/w3c-logo.png)"></div>
```

#### Failed Example 3

This `img` element inside a `div` positioned off screen has an empty [accessible name][] and an [implicit role][] of `img`.

```html
<div style="margin-left:-9999px;"><img src="/test-assets/shared/w3c-logo.png" /></div>
```

#### Failed Example 4

> This `img` element has an empty [accessible name][] because the space in the `alt` attribute is trimmed off by the [accessible name computation](https://www.w3.org/TR/accname-1.1/).

Reviewer 2: Do we want to expand the description to say that even if it is trimmed by accname computation, it still prevent `alt` to be empty for role mappings?

Author: Yeah, good idea

```html
<img src="/test-assets/shared/w3c-logo.png" alt=" " />
```

#### Failed Example 5

This `img` element has an [explicit role][] of `none`. However, it is [focusable][] due to the `tabindex` attribute. Because of this it has a [semantic role][] of `img` due to [Presentational Roles Conflict Resolution][]. It does not have an accessible name.

```html

> <img role="none" tabindex="0" />

Reviewer 2: ```suggestion
<img role="none" tabindex="0" src="/test-assets/shared/w3c-logo.png"/>
```
Let's keep on with having actual images in the examples 😃

Author: oh whoops! Thanks! Seems I forgot to review this one myself before I put it out.

#### Inapplicable Example 1

> The `svg` element has an [implicit role][] of `graphic-document`.

Reviewer: ```suggestion
This `svg` element has an [implicit role][] of `graphics-document`.
```

```html
<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow" />
</svg>
```

#### Inapplicable Example 2

This element has a [semantic role][] of `img`, but is not [included in the accessibility tree][].

```html

> <div role="img" aria-hidden="true"></div>

Reviewer 2: ```suggestion
<div role="img" aria-hidden="true" style="width:72px; height:48px; background-image: url(/test-assets/shared/w3c-logo.png)"></div>
```

Author: sure, no objection.

```

#### Inapplicable Example 3

This `img` element is not [included in the accessibility tree][].

```html
<img src="/test-assets/shared/w3c-logo.png" aria-hidden="true" />
```

#### Inapplicable Example 4

> This element is not an `img` element.

Reviewer 2: ```suggestion
This element is neither an `img` element nor has a role of `img`.
```
